### PR TITLE
feat: added s3 bucket util library.

### PR DESCRIPTION
**Pull request by krackerman**

### CODE REVIEW

File: src/module/s3.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+module.exports = ({ call, getService }) => {
+  const putGzipObject = ({ bucket, key, data }) => call('s3:putObject', {
+    ContentType: 'application/json',
+    ContentEncoding: 'gzip',
+    Bucket: bucket,
+    Key: key,
+    Body: zlib.gzipSync(data, { level: 9 })
+  });
+
+  const getGzipObject = ({ bucket, key, expectedErrorCodes }) => call(
```
Comment (simlu) [inline src/module/s3.js]: start working on readme

File: test/module/s3.spec.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
+const expect = require('chai').expect;
+const { describe } = require('node-tdd');
+const index = require('../../src');
+
+describe('Testing s3 Util', {
+  useNock: true,
+  envVarsFile: 'config.env.yml',
```
[the remark this answers is not in the document]
Comment (krackerman) [reply test/module/s3.spec.js]: I have not addressed yet. What do you mean by the default file?

File: README.md [hunk 3]
```diff
@@ -84,6 +84,26 @@ The exposed `ingest` method should only be used to seed the queue. Messages gene
...
+Retrieves only the metadata from an object in an Amazon S3 bucket. Uses [s3:headObject](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#headObject-property).
+
+#### s3.deleteObject({ bucket: String, key: String })
+Delete object from an Amazon S3 bucket at key. Uses [s3:deleteObject](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#deleteObject-property).
+
+#### s3.listObjects({ bucket: String, limit: Number, startAfter: String })
+List objects keys in an Amazon S3 bucket. Internally this pages until the 
+limit is reached or no more keys are available. Uses [s3:listObjectsV2](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#listObjectsV2-property).
+
+#### s3.escapeKey(key: String)
```
Comment (simlu) [inline README.md]:
```suggestion
#### s3.decodeKey(key: String)
```

File: src/module/s3.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    { expectedErrorCodes }
+  ).then((r) => (expectedErrorCodes.includes(r) ? r : JSON.parse(zlib.gunzipSync(r.Body).toString('utf8'))));
+
+  const headObject = ({ bucket, key, expectedErrorCodes }) => call(
+    's3:headObject',
+    { Bucket: bucket, Key: key },
+    { expectedErrorCodes }
+  );
+
+  const deleteObject = ({ bucket, key }) => call('s3:deleteObject', { Bucket: bucket, Key: key });
```
Comment (simlu) [inline src/module/s3.js]:
```suggestion
  const deleteObject = ({ bucket, key, expectedErrorCodes }) => call('s3:deleteObject', { Bucket: bucket, Key: key }, { expectedErrorCodes });
```

**Pull request merged**